src/core/manager/mod.rs

```diff
@@ -782,6 +792,10 @@ impl<X: XConn> WindowManager<X> {
     /// Cycle between known [screens][Screen]. Does not wrap from first to last
     pub fn cycle_screen(&mut self, direction: Direction) -> Result<()> {
         let actions = self.state.screens.cycle_screen(direction, &self.conn)?;
+
+        let wix = self.screens.active_ws_index();
+        self.conn.set_current_workspace(wix)?;
```

Comment:
This is redundant as it you are already handling this in `focus_workspace` which will be called as a result of the event actions here

Reply:
I think the `Screens::cycle_screen` changes the screen, causing `focus_workspace` to exit early in the first if-test and not handling it. I will try to write some tests to confirm this and not having to rely on my manual testing.